src/Amazon.Common.DotNetCli.Tools/Utilities.cs

```diff
@@ -208,11 +209,72 @@ public static string DeterminePublishLocation(string workingDirectory, string pr
         public static string LookupTargetFrameworkFromProjectFile(string projectLocation)
         {
             var projectFile = FindProjectFileInDirectory(projectLocation);
+            if (string.IsNullOrEmpty(projectFile))
+            {
+                throw new FileNotFoundException("Could not find a project file in the specified directory.");
+            }
 
-            var xdoc = XDocument.Load(projectFile);
+            var arguments = new[]
```

Comment:
We look up `TargetFramework`, `OutputType`, and `PublishAot` from the project file in this class. I would like to see this logic generalized so that we can have a method that takes in the project and a `params` of property names and returns back a dictionary of values. I don't really want 3 different code paths starting the `dotnet` process which will end up drifting.

The shared method should have the fallback logic I describe below that if the process fails parse the project xml like it does today so we don't risk any breaking behavior.

Reply:
updated in latest commit